### ISSUE TITLE
Warning pop to make scams a little harder

### ISSUE THREAD

**Issue by Darin755** [rustdesk/rustdesk#1675]
Is there any plans to add warning popup that warns the user that it could be a scam? It seems like this software could easily be abused for malicious intentions. It would be nice to have a simple warning popup that has a brief explanation on the danger of blindly accepting a connection.

**Comment by GameParrot** [comment]
Maybe add a warning that it could be a scam if someone from India is trying to connect (because most scammers are in India).

**Comment by Darin755** [comment]
> Maybe add a warning that it could be a scam if someone from India is trying to connect (because most scammers are in India).

Isn't that racist?

**Comment by bmcgonag** [comment]
I understand the desire to warn someone about potential scams, but any software could be abused in just such a way.  I think a well worded warning might be useful a very small percentage of the time, but generally if someone is being scammed, they have been contacted by phone or some other means, and asked to install the application, and thus the scammer would simply tell them there is a warning, and they should ignore it unless anyone else tries to connect.

**Comment by GameParrot** [comment]
Another idea could be to always show the IP address of the remote user when they connect. This would make scammers not want to use the software.

**Comment by Darin755** [comment]
I was thinking a popup that says "You are about to give someone access to your computer. Proceed?". It probably should be worded to be a little more intimidating but you get the picture.

**Comment by GameParrot** [comment]
Another idea is it could say something like, "WARNING: YOU MAY BE BEING SCAMMED!
If you are on the phone with someone who asked you to install RustDesk, do not install and hang up immediately. They are likely a scammer trying to steal your money or other private information." during installation. The installer could also make you check a box saying that your were not asked to install RustDesk by someone you don't know on the phone.

**Comment by Darin755** [comment]
> Another idea is it could say something like, "WARNING: YOU MAY BE BEING SCAMMED! If you are on the phone with someone who asked you to install RustDesk, do not install and hang up immediately. They are likely a scammer trying to steal your money or other private information." during installation. The installer could also make you check a box saying that your were not asked to install RustDesk by someone you don't know on the phone.

I personally like your idea of a checkbox. It makes them slow down and read

**Comment by dzatoah** [comment]
Maybe we could say something like: "A legitimate remote supporter would never ask for login credentials, especially not  bank accounts or online payment systems. Never share your login credentials with anyone you don't trust 100% and know personally! If you are in doubt please ask your local Administrator before proceeding."

**Comment by rustdesk** [comment]
👍 
@21pages Let's think about where to put such message.

**Comment by dzatoah** [comment]
I would suggest to put this here:
![image](https://user-images.githubusercontent.com/16420080/222706938-b3f6855d-1726-43ed-8448-d73e6d4057c2.png)


Or maybe a dialog popup if you click on the accept button.

**Comment by rustdesk** [comment]
> on the accept button.

And with a checkbox `don't show it any more for dainel's requrest` or `don't show it any more for all requests`?

**Comment by rustdesk** [comment]
@dinger1986 @grummbeer do you have any idea about this?

pop uo a dialog when first time running RustDesk, or for each connection?

**Comment by dinger1986** [comment]
Yeah I think a pop up.

Also on the website

`Something like Warning! Remote control software like RustDesk is commonly used by scammers, make sure whoever you are giving your is trustworthy`

**Comment by rustdesk** [comment]
> website

https://github.com/rustdesk/rustdesk/discussions/3669

**Comment by grummbeer** [comment]
Important.

Some thoughts

- Messages along the way the scammer leads the victim. Starts with the download/website.
- The danger depends on "Permissions" settings. 
  - Use "screen share" as default
  - place a message there, above the checkboxes if other than "screen share" is selected
- Destructive actions need a extra confirmation. Granting access can also be destructive. So show an confirmation dialog on every/each "Accept"
   - Add checkox to this dialog "trust this [ID]" 
      - name fakeable?
      - IDs could exists twice? 
   - The "trust" must be revokable, in settings/somewhere at peers
- React to settings
   - Show permissions in confirmations dialog
   - If is only "screen shareing" > [visual indicator] "it is safe to accept", true?

**Comment by grummbeer** [comment]
True. 

It's about finding the right balance. Having it but not being annoying or mess up the UI.

We're not talking about nerds oder admins. I don't know exactly, but I think for a few years now most users are "common" users in home offices who click as soon as a "click there" reaches their ear.

**Comment by dinger1986** [comment]
The biggest risk isn't users in a business it's home users. The granny and grandads who don't know any better.

Some warnings will help for sure.

Even on the download screen and a count down so there's a delay and time for them to think?

**Comment by grummbeer** [comment]
> countdown

After the countdown the download starts automatically :-)

**Comment by dinger1986** [comment]
The issue isn't the download though. So a countdown with that warning on the screen would be fine.

The reality is if there's enough warnings bad actors just simply won't use it for risk of being found out!

**Comment by dinger1986** [comment]
The issue isn't the download though. So a countdown with that warning on the screen would be fine.

The reality is if there's enough warnings bad actors just simply won't use it for risk of being found out!

**Comment by grummbeer** [comment]
> The reality is if there's enough warnings bad actors just simply won't use it for risk of being found out!

Not that bad, i think.

Who are the audience for such kind of messages?

- People who do not know at all what a "remote desktop" is or how it could harm them.
- Not people who are already experienced in getting help via remote desktop or who themselves control their workstations in the company via home office.

Right?

**Comment by dinger1986** [comment]
Yeah the audience is people who are likely to be scammed not professionals

**Comment by grummbeer** [comment]
> pop up a dialog when first time open RustDesk, or for each connection?

Would say website and on the first launch.

**Comment by dinger1986** [comment]
Yeah agreed!

**Comment by rustdesk** [comment]
👍

**Comment by dinger1986** [comment]
need to find the balance remember!

**Comment by Darin755** [comment]
I think it be a good idea to limit the number of connection attempts in a 5min period.

If the popup keeps coming up people may accept it just to make it go away

**Comment by dinger1986** [comment]
Yes limit the connections would be good as well.

The main thing is to make bad actors not use rustdesk. Unfortunately there's no way to stop them doing what they are doing!!

**Comment by dinger1986** [comment]
Yes limit the connections would be good as well.

The main thing is to make bad actors not use rustdesk. Unfortunately there's no way to stop them doing what they are doing!!

**Comment by test2a** [comment]
I have a proposition to make. There are dozens of anti scam YouTubers and on twitch. They are the real experts in both knowing what the target audience (older gen ornon technical people) language they might understand or how scammers work. 

My suggestion is, we, the community reach out to them and ask for their advice. They would surely be able to help in the ux part. 

Otherwise we are merely splitting hair discussing what should not what should not be because we simply don't know and are guessing.

**Comment by Darin755** [comment]
> I have a proposition to make. There are dozens of anti scam YouTubers and on twitch. They are the real experts in both knowing what the target audience (older gen ornon technical people) language they might understand or how scammers work.
> 
> My suggestion is, we, the community reach out to them and ask for their advice. They would surely be able to help in the ux part.
> 
> Otherwise we are merely splitting hair discussing what should not what should not be because we simply don't know and are guessing.

This is a really good idea. I would also try Reddit
